### PR TITLE
Several small fixes to mixer envoy ext authz and gRPC access log API support

### LINKED ISSUE #23580: Provide a shim to convert OOP adapter to use Envoy native API
As part of efforts for Mixer deprecation, we need a way for current out of process (OOP) adapter user to migrate off Mixer template based API. The long term solution is to convert out of process adapter to a Wasm filter. However this process could be time consuming and requires Wasm runtime to be production ready, which is not guaranteed to happen on 1.7.

As an interim step, a shim could be provided so that OOP adapter user could compile that with their adapter, which:

- Consumes native Envoy extern-authz API for policy check and access log API for telemetry.
- Is generated based on given Mixer instances/handlers/template/rules/adapter.
- Does not support dynamic configuration.

Guide will be provided on istio.io about how to migrate off out of process adapter to use this shim generation tool. We also need to be very clear that this shim generator should only be considered as a short term solution for transition to Wasm and will only be supported until Wasm production ready release or the release after that.

@mandarjog @douglas-reid

### PR DESCRIPTION
**Pull request by bianpengyuan**
Fixes #23580 with wiki page guide: https://github.com/istio/istio/wiki/Enabling-Envoy-Authorization-Service-and-gRPC-Access-Log-Service-With-Mixer

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: mixer/pkg/attribute/envoyProtoBag.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func stripPrincipal(principal string) string {
 func AuthzProtoBag(req *authz.CheckRequest) *EnvoyProtoBag {
 	pb := envoyProtoBags.Get().(*EnvoyProtoBag)
 	reqMap := make(map[string]interface{})
-	reqMap["context.reporter.kind"] = "inbound"
+	reqMap["context.reporter.kind"] = "unknown"
```
Comment (douglas-reid) [inline mixer/pkg/attribute/envoyProtoBag.go:95]: The `ExtAuthz::check` should be `inbound` no matter what, right?  what is gained by setting it to `unknown`?

Comment (bianpengyuan) [reply mixer/pkg/attribute/envoyProtoBag.go:95]: It is actually `outbound` at ingress gateway, so I updated it to `unknown`. I don't expect that anyone would reference this attribute in their check instance, so I am fine to use either `unknown` or `inbound`.

File: mixer/pkg/attribute/envoyProtoBag.go [hunk 2]
```diff
@@ -173,7 +173,7 @@ func AccessLogProtoBag(msg *accesslog.StreamAccessLogsMessage, num int) *EnvoyPr
 			reqMap["connection.sent.bytes"] = int64(connection.GetSentBytes())
 		}
 	}
-	reqMap["context.reporter.kind"] = "inbound"
+	reqMap["context.reporter.kind"] = "unknown"
```
Comment (douglas-reid) [inline mixer/pkg/attribute/envoyProtoBag.go:176]: iirc, we used to have rule matches based on inbound/outbound (client/server side metrics, etc.). is it ok to not try to figure out the kind here?

File: mixer/pkg/attribute/envoyProtoBag.go (hunk 1 above)
Comment (douglas-reid) [reply mixer/pkg/attribute/envoyProtoBag.go:95]: gateway behavior bites again. the issue will be if anyone is using APAs or similar that are selected based on inbound/outbound.  but `unknown` seems fine to start with, i guess.

File: mixer/pkg/attribute/envoyProtoBag.go (hunk 2 above)
Comment (bianpengyuan) [reply mixer/pkg/attribute/envoyProtoBag.go]: Yeah, I cannot think of a way to distinguish this. Now I think about this, it seems like it indeed bring not much value making it `unknown`. The limitation has already been documented pretty clear, so making it `inbound` might save a configuration change for users. I reverted back this change.

**Comment by JimmyCYJ** [comment]
Is this PR ready? @douglas-reid could you take a look?

**Comment by bianpengyuan** [comment]
No, I think this is the last PR needed for this Mixer shim layer feature so I attached the release note with this. Do you want em to submit a separate PR for the release note?

**Review by jwendell — APPROVED** [review]
@bianpengyuan Fair enough, thanks.

**Pull request merged**